### PR TITLE
feat: use contentful for vacancies

### PR DESCRIPTION
**Pull request by hofmann-felix**
### What's included?
* Our vacancies now no longer come from personio but from contentul. A new content model was implemented: Bildschirm­foto 2023-03-14 um 09 09 41
* The four jobs on satellytes.com were added to contentful
* The vacancies in the FE, structured data and image card generation have been adapted to contentful
* All code related to Personio has been deleted (exception: Personio API to submit applications from the form)

### Preview:
* https://satellytescommain21751-featcontentfulvacancies.gatsbyjs.io/career/

### CODE REVIEW

**Comment by georgiee** [comment]
Nice, looking forward completing the transition away from Personio with that last missing bit 🙌

File: gatsby/util/extract-personio-slug.ts [hunk 1]
```diff
@@ -1,21 +0,0 @@
-export const PERSONIO_SLUG_FIELD_NAME = 'Slug';
-
-/**
```
Comment (hofmann-felix) [inline gatsby/util/extract-personio-slug.ts:3]: I found no usage of this file and therefore deleted it. Also deleted is the `sy-personio-jobs-source-plugin`

Comment (georgiee) [reply gatsby/util/extract-personio-slug.ts:3]: yes please, we don't care for Personio anymore 😌

File: src/components/pages/career-details/career-details-structured-data.tsx [hunk 2]
```diff
@@ -27,12 +21,9 @@ const normalizeJobScheduleFormat = (schedule: string): string => {
...
  */
-const generateHtmlDescription = (position: SyPersonioJob): string => {
-  const htmlIntro = `${position.short} <br><br>`;
-  const htmlSections = position.sections
-    .map((section) => `${section.headline} <br> ${section.descriptionHtml}`)
-    .join('<br><br>');
-  return htmlIntro + htmlSections;
+const generateHtmlDescription = (position: ContentfulVacancy): string => {
+  const richText = renderRichText(position.content, {});
+  return renderToString(<>{richText}</>);
```
Comment (hofmann-felix) [inline src/components/pages/career-details/career-details-structured-data.tsx]: According to the [Google Guide](https://developers.google.com/search/docs/appearance/structured-data/job-posting#job-posting-definition) for job descriptions, only certain html tags are allowed. Other tags are apparently simply ignored ("_While the tags won't affect the formatting in the feature, you can safely include them on the page._"). Before the changes, there were only supported tags in the description. I'm not quite sure if it makes sense to parse the description exactly as before or leave it as it is now (with unsupported tags like h1). 
All other fields that are relevant for Google didn't change (tested with https://search.google.com/test/rich-results)


<details>
  <summary>Google Job Description (Before)</summary>
Bildschirm­foto 2023-03-14 um 08 45 06
</details>

<details>
  <summary>Google Job Description (Updated)</summary>
Bildschirm­foto 2023-03-14 um 09 04 16


</details>

File: src/components/pages/career-details/career-details-structured-data.tsx [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
 }
 
-/**
- * Convert the kebab-case, lowercase Personio values like `full-time`
- * to the expected snake_case, uppercase format `FULL_TIME`.
- */
-const normalizeJobScheduleFormat = (schedule: string): string => {
-  return schedule.toUpperCase().replace('-', '_');
-};
-
```
Comment (hofmann-felix) [inline src/components/pages/career-details/career-details-structured-data.tsx:17]: The contentful model only allows `FULL_TIME` or `PART_TIME`

File: gatsby-config.ts [hunk 4]
```diff
@@ -43,12 +43,6 @@ const gatsbyConfig: GatsbyConfig = {
     siteUrl: BASE_URL,
   },
   plugins: [
-    {
-      resolve: `sy-personio-jobs-source-plugin`,
```
Comment (georgiee) [inline gatsby-config.ts:47]: bye 👋

File: gatsby/create-pages/create-career-pages.ts [hunk 5]
```diff
@@ -5,92 +5,71 @@ const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
   `${process.cwd()}/src/templates/career-details.tsx`,
 );
 
-/**
- * There can be multiple positions with the same jobId but a different language.
- * This methods helps to find the (first) position with the same jobId as the given jobId.
- */
-const createComplementFinder = (positions) => {
```
Comment (georgiee) [inline gatsby/create-pages/create-career-pages.ts:12]: finally gone, thanks for removing it

File: src/components/pages/career-details/career-details.stories.mdx [hunk 6]
```diff
@@ -8,27 +7,6 @@ import { Form } from './new-career-form/career-form';
 
 Here the parts to build the career details page.
 
-## Job Description
```
Comment (georgiee) [inline src/components/pages/career-details/career-details.stories.mdx:11]: Do you use the stories? I'm happy to remove them if nobody uses them. It's maybe a little bit over the edge for us and was good in the beginning as a playground. But whenever I see stories these days it feels like outdated, unused, obsolete.

**Review by georgiee — APPROVED** [review]
Wow. That couldn't have been done anything better  ✨

That was a flawless review experience. You just took care of everything.
Plus you gave me that relaxing feeling of removing more from that ugly Personio related workaround stuff. 

The diff stats speaks for itself.
Screenshot 2023-03-15 at 13 24 47


Thank you for making this happen 😌 Now let's merge 🤩

**Pull request merged**